service/src/main/java/com/commercetools/pspadapter/payone/PaymentHandler.java

```diff
@@ -0,0 +1,111 @@
+package com.commercetools.pspadapter.payone;
+
+import com.commercetools.pspadapter.payone.domain.ctp.CommercetoolsQueryExecutor;
+import com.commercetools.pspadapter.payone.domain.ctp.PaymentWithCartLike;
+import com.commercetools.pspadapter.payone.domain.ctp.exceptions.NoCartLikeFoundException;
+import io.sphere.sdk.client.ErrorResponseException;
+import io.sphere.sdk.client.NotFoundException;
+import io.sphere.sdk.http.HttpStatusCode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.ConcurrentModificationException;
+import java.util.concurrent.CompletionException;
+
+import static com.commercetools.pspadapter.tenant.TenantLoggerUtil.createLoggerName;
+import static io.sphere.sdk.http.HttpStatusCode.INTERNAL_SERVER_ERROR_500;
+import static java.lang.String.format;
+import static java.util.Optional.ofNullable;
+
+public class PaymentHandler {
+
+    /**
+     * How many times to retry if {@link ConcurrentModificationException} happens.
+     */
+    private static final int RETRIES_LIMIT = 20;
+    private static final int RETRY_DELAY = 100; // msec
```

Comment:
`RETRY_DELAY` is applied to `ConcurrentModificationException` exceptions, not to HTTP responses.